cdap-data-fabric/src/main/java/io/cdap/cdap/data2/dataset2/lib/table/MDSKey.java

```diff
@@ -97,6 +97,14 @@ public String getString() {
       return Bytes.toString(getBytes());
     }
 
+    /**
+     * @throws BufferUnderflowException if there is no boolean as expected
+     * @return the next boolean part in the splitter
+     */
+    public boolean getBoolean() {
+      return Bytes.toBoolean(getBytes());
```

Comment:
This is not correct, getBytes() will read an int, then read the next X bytes. It is meant to be used when reading a part of the key that was written using `add(byte[] part)`. The `add(boolean part)` method you added just appends a single byte, which means this method should read the next byte from the byteBuffer and return Bytes.toBoolean() on it.

Please add some unit tests for these new methods.